JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java

```diff
@@ -78,35 +80,52 @@ public void setKBClient(TripleStoreClientInterface kbClient) {
     	this.rdfClient.setKBClient(kbClient);
     }
 
-    /**
-     * Initialise time series in triple store and relational database
-     * @param dataIRIs list of dataIRIs as Strings
-     * @param dataClass list of data classes for each dataIRI
-     * @param timeUnit time unit as (full) IRI
+	/**
+	 * Initialise time series in triple store and relational database
+	 * @param dataIRIs list of dataIRIs as Strings
+	 * @param dataClass list of data classes for each dataIRI
+	 * @param timeUnit time unit as (full) IRI
 	 * @param conn connection to the RDB
-     */
-    public void initTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn) {
+	 * @param type type of TimeSeries data to be instantiated. (optional)
+	 *             Allowed values:
+	 *             https://www.theworldavatar.com/kg/ontotimeseries/StepwiseCumulative,
+	 *             https://www.theworldavatar.com/kg/ontotimeseries/CumulativeTotal,
+	 *             https://www.theworldavatar.com/kg/ontotimeseries/Instantaneous,
+	 *             https://www.theworldavatar.com/kg/ontotimeseries/Average
+	 *             If not specified, default value: TimeSeries
+	 * @param duration Required for Average Time Series. Numeric duration of the averaging period for Average TimeSeries of type Duration. Only positive values are allowed. (optional)
+	 * @param unit Required for Average Time Series. Temporal unit type of the averaging period for Average TimeSeries. (optional)
+	 *             Allowed values of type ChronoUnit:
+	 *             ChronoUnit.SECONDS, ChronoUnit.MINUTES, ChronoUnit.HOURS, ChronoUit.DAYS, ChronoUnit.WEEKS, ChronoUnit.MONTHS, ChronoUnit.YEARS
+	 *
+	 */
+	public void initTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn, String type, Duration duration, ChronoUnit unit) {
+		String tsIRI;
+		// Create random time series IRI in the format: <Namespace><ClassName>_<UUID>
+		if (type.equals(null)){
+			tsIRI = TimeSeriesSparql.ns_kb + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.StepwiseCumulative)){
+			tsIRI = TimeSeriesSparql.StepwiseCumulative + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.CumulativeTotal)){
+			tsIRI = TimeSeriesSparql.CumulativeTotal+  "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.Instantaneous)){
+			tsIRI = TimeSeriesSparql.Instantaneous + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(TimeSeriesSparql.Average)){
+			tsIRI = TimeSeriesSparql.Average + "Timeseries_" + UUID.randomUUID();
+		}
+		else {
+			throw new JPSRuntimeException(exceptionPrefix + "TimeSeries type: " + type + " is invalid");
+		}
```

Comment:
Avoid these sort of if-else codes

Reply:
should I replace it with switch case? if not, what would be the alternative?

Reply:
see comment below, have different methods, initAverageTimeSeries, this is what I prefer in this case. Alternative will be to use enums